### PR TITLE
Compile fix for OSX

### PR DESCRIPTION
**Pull request by hgaiser**
This adds a number of checks to make the SDK compile under OSX.

ps. I didn't bother with the world demo. Should it not perhaps just be deleted?

### CODE REVIEW

**Comment by wjwwood** [comment]
This looks reasonable to me, but I don't have time to test it. It should be tested on OS X and Linux since this would likely effect the Linux build too.

**Comment by hgaiser** [comment]
I will check it soon on Linux to see if it still compiles and works there.

**Comment by hgaiser** [comment]
It still compiles on linux, but I didn't get it to work on that laptop. I couldn't get it to work with either the original version or with this branch, so thats another issue I guess, not related to these changes.. However since it compiled and I only edited cmake files, I can say with almost absolute guarantee that it will work as it did before :p

**Comment by wjwwood** [comment]
Looks good to me, but we should get a review from another maintainer as well. @dgossow @tfoote

File: Samples/OculusWorldDemo/CMakeLists.txt [hunk 1]
```diff
@@ -4,28 +4,45 @@ INCLUDE_DIRECTORIES(../../LibOVR/Include ../../LibOVR/Src)
...
   Player.cpp
   ../../Samples/CommonSrc/Platform/Platform.cpp
-  ../../Samples/CommonSrc/Platform/Linux_Platform.cpp
-  ../../Samples/CommonSrc/Platform/Linux_Gamepad.cpp
   ../../Samples/CommonSrc/Render/Render_Device.cpp
   ../../Samples/CommonSrc/Render/Render_GL_Device.cpp
   ../../Samples/CommonSrc/Render/Render_LoadTextureDDS.cpp
   ../../Samples/CommonSrc/Render/Render_LoadTextureTGA.cpp
   ../../Samples/CommonSrc/Render/Render_XmlSceneLoader.cpp
+  ${PLATFORM_FILES}
```
Comment (tfoote) [inline Samples/OculusWorldDemo/CMakeLists.txt]: This like looks like it should be PLATFORM_SOURCE_FILES

**Comment by tfoote** [comment]
One comment to address otherwise it looks reasonable.

**Comment by dgossow** [comment]
lgtm

**Comment by hgaiser** [comment]
Change added, though I must stress again that I disabled building of that sample, I could not get it to compile and felt it was unnecessary to spend time on it..

For references:

```
Undefined symbols for architecture x86_64:
  "OVR::Platform::Application::CreateApplication()", referenced from:
      -[OVRApp run] in OSX_Platform.mm.o
  "OVR::Platform::Application::DestroyApplication(OVR::Platform::Application*)", referenced from:
      -[OVRApp run] in OSX_Platform.mm.o
```

**Comment by wjwwood** [comment]
Ok, lgtm too. I'll merge it.

**Comment by hgaiser** [comment]
Great! https://github.com/ros-visualization/oculus_rviz_plugins/pull/17 is connected to this commit by the way.

**Pull request merged**